tests/components/weatherflow/conftest.py

```diff
@@ -0,0 +1,151 @@
+"""Fixtures for IntelliFire integration tests."""
+import asyncio
+from collections.abc import Generator
+import json
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED
+from pyweatherflowudp.const import DEFAULT_HOST
+from pyweatherflowudp.device import WeatherFlowDevice
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+
+from homeassistant.components.weatherflow.const import DOMAIN
+from homeassistant.const import CONF_HOST
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock, None, None]:
+    """Mock setting up a config entry."""
+    with patch(
+        "homeassistant.components.weatherflow.async_setup_entry", return_value=True
+    ) as mock_setup:
+        yield mock_setup
+
+
+@pytest.fixture
+def mock_config_entry() -> MockConfigEntry:
+    """Return a mock config entry."""
+    return MockConfigEntry(domain=DOMAIN, data={CONF_HOST: "1.2.3.4"})
+
+
+@pytest.fixture
+def mock_config_entry2() -> MockConfigEntry:
+    """Return a mock config entry."""
+    return MockConfigEntry(domain=DOMAIN, data={CONF_HOST: DEFAULT_HOST})
+
+
+@pytest.fixture
+def mock_has_devices() -> Generator[AsyncMock, None, None]:
+    """Return a mock has_devices function."""
+    with patch(
+        "homeassistant.components.weatherflow.config_flow._async_can_discover_devices",
+        return_value=True,
+    ) as mock_has_devices:
+        yield mock_has_devices
+
+
+@pytest.fixture
+def mock_has_no_devices() -> Generator[AsyncMock, None, None]:
+    """Return a mock has_devices function returning False."""
+    with patch(
+        "homeassistant.components.weatherflow.config_flow._async_can_discover_devices",
+        return_value=False,
+    ) as mock_has_devices:
+        yield mock_has_devices
+
+
+@pytest.fixture
+def mock_has_devices_error_listener() -> Generator[AsyncMock, None, None]:
+    """Return a mock has_devices returning an error."""
+    with patch(
+        "homeassistant.components.weatherflow.config_flow._async_can_discover_devices",
+        side_effect=ListenerError,
+    ) as mock_has_devices:
```

Comment:
Please patch the pypi code and not the integration code `(_async_can_discover_devices`).